src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -1299,9 +1299,44 @@ public sealed class SocketsHttpHandlerTest_AutoRedirect : HttpClientHandlerTest_
         public SocketsHttpHandlerTest_AutoRedirect(ITestOutputHelper output) : base(output) { }
     }
 
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
     public sealed class SocketsHttpHandler_DefaultCredentialsTest : DefaultCredentialsTest
     {
         public SocketsHttpHandler_DefaultCredentialsTest(ITestOutputHelper output) : base(output) { }
+
+        [Fact]
+        public async Task SocketsHttpHandler_UseDefaultCredentials_OneRequestOnlyForBasicAuth()
+        {
+            var requestCount = 0;
+            await LoopbackServerFactory.CreateClientAndServerAsync(
+                async url =>
+                {
+                    using (var handler = new SocketsHttpHandler())
+                    using (var invoker = new HttpMessageInvoker(handler))
+                    {
+                        handler.Credentials = CredentialCache.DefaultCredentials;
+                        var request = new HttpRequestMessage(HttpMethod.Get, url);
+                        var response = await invoker.SendAsync(request, CancellationToken.None);
+                        Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+                    }
+                },
+                async server =>
+                {
+                    var responseHeader = new[] { new HttpHeaderData("WWW-Authenticate", "Basic realm=\"Test Realm\"") };
+                    await server.HandleRequestAsync(HttpStatusCode.Unauthorized, responseHeader);
+                    requestCount++;
+
+                    // Only one request should be sent.
+                    await IgnoreExceptions(async () => 
+                    {
+                        await server.HandleRequestAsync(HttpStatusCode.Unauthorized, responseHeader).WaitAsync(TimeSpan.FromSeconds(10));
+                        requestCount++;
+                    });
+                }
+            );
+
+            Assert.Equal(1, requestCount);
```

Comment:
```suggestion
            await LoopbackServerFactory.CreateClientAndServerAsync(
                async url =>
                {
                    using var handler = new SocketsHttpHandler();
                    using var client = new HttpClient(handler);
                    handler.Credentials = CredentialCache.DefaultCredentials;

                    using HttpResponseMessage response = await invoker.GetAsync(url);
                    Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
                },
                async server =>
                {
                    var responseHeader = new[] { new HttpHeaderData("WWW-Authenticate", "Basic realm=\"Test Realm\"") };
                    await server.HandleRequestAsync(HttpStatusCode.Unauthorized, responseHeader);
                }
            );
```

It's okay to validate that a single request was sent just by the fact that the request succeeded here.
We don't want to make the test take 10 seconds every time in order to hit a timeout.